### PR TITLE
Create UI Component test framework

### PR DESCRIPTION
**Pull request by PhysicsJon**
Move functionality to new library
Add new project for UI Component tests
Add Tests and structure to project
Add pipeline task to run tests on build

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/NHSD.GPIT.BuyingCatalogue.WebApp.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <CopyRefAssembliesToPublishDirectory>false</CopyRefAssembliesToPublishDirectory>
-    <UserSecretsId>8b1e8b83-a66c-4c7c-bc90-03575c960367</UserSecretsId>
+    <UserSecretsId>npm </UserSecretsId>
```
Comment (bjss-colin) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/NHSD.GPIT.BuyingCatalogue.WebApp.csproj:6]: Looks like a possible typing error. Needs to be reverted back to being the GUID

File: src/NHSD.GPIT.BuyingCatalogue.UI.Components.WebApp/wwwroot/ts/SupplierManagement.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿window.onload = function () {
```
Comment (bjss-colin) [inline src/NHSD.GPIT.BuyingCatalogue.UI.Components.WebApp/wwwroot/ts/SupplierManagement.ts:1]: Should these be brought over to the UI Components project or can they be tested in the web app? They are specific to the web app as it deals directly with supplier data.

**Review by bjss-colin — COMMENTED** [review]
For the most part it looks fine, there's just a few TypeScript and node related files which I'm not sure if the Components UI project is the right place for them.

File: src/NHSD.GPIT.BuyingCatalogue.UI.Components.WebApp/wwwroot/ts/SupplierManagement.ts (hunk 2 above)
Comment (PhysicsJon) [reply src/NHSD.GPIT.BuyingCatalogue.UI.Components.WebApp/wwwroot/ts/SupplierManagement.ts:1]: It's a bit of a nightmare as there are loads of typescript errors without the tsconfig file, if that's added without any ts files of any type it fails. I suppose we could have a simple ts file that does nothing, but I've not had the chance to look at that so far

Comment (PhysicsJon) [reply src/NHSD.GPIT.BuyingCatalogue.UI.Components.WebApp/wwwroot/ts/SupplierManagement.ts]: I've replaced this file with a mostly blank ts file

**Review by bjss-colin — APPROVED** [review]

**Pull request merged**